aten/src/ATen/native/quantized/cpu/quant_utils.h

```diff
@@ -136,4 +165,20 @@ static torch::List<int64_t> MakeArgForConv1d(const torch::List<int64_t>& arg,
   return result;
 }
 
+// The range for using FP16 quantization of weights requires that the elements
+// should be in the range of [5.96e-8, 65504]. If it is out of range, then the
+// number will be saturated to max or min representable values by FP16.
+inline void HandleWeightsSaturation(int64_t N, float* weight) {
+  const float kFp16Max = RawUint16ToFp16(0x7BFF);
+  bool found_out_of_range = false;
+  for (int64_t i = 0; i < N; ++i) {
+    if (CheckAndSaturate<float>(kFp16Max, weight + i)) {
```

Comment:
optional nit: calling a function with side effects inside the if condition is usually not intuitive.  Maybe we can move the calculation into a bool before the if block?